PhysicsTools/Heppy/src/IsolationComputer.cc

```diff
@@ -129,6 +158,43 @@ float heppy::IsolationComputer::isoSumRaw(const std::vector<const pat::PackedCan
     return isosum;
 }
 
+reco::Candidate::LorentzVector heppy::IsolationComputer::isoP4Raw(const std::vector<const pat::PackedCandidate *> & cands, const reco::Candidate &cand, float dR, float innerR, float threshold, SelfVetoPolicy selfVeto, int pdgId) const 
```

Comment:
If these are copies of the non-P4 methods except that they add up cand->p4() rather than cand->pt(), then it would be better to make a template and call it (just like the `algoWithin` in DataFormats/RecoCandidate/interface/IsoDeposit.h )